packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/ConstructorProvider.cs

```diff
@@ -57,5 +57,31 @@ public ConstructorProvider(ConstructorSignature signature, ValueExpression bodyE
                 ? MethodProviderHelpers.BuildXmlDocs(signature.Parameters, signature.Description, null, null)
                 : null);
         }
+
+        public void Update(
+            MethodBodyStatement? bodyStatements = null,
+            ConstructorSignature? signature = null,
+            ValueExpression? bodyExpression = null,
+            XmlDocProvider? xmlDocs = null)
+        {
+            if (signature != default)
+            {
+                Signature = signature;
+            }
+            if (bodyExpression != null)
+            {
+                BodyExpression = bodyExpression;
+                BodyStatements = null;
+            }
+            if (bodyStatements != default)
+            {
+                BodyStatements = bodyStatements;
+                BodyExpression = null;
+            }
+            if (xmlDocs != default)
+            {
+                XmlDocs = xmlDocs;
+            }
```

Comment:
could we use `null` for all of them? It is kind of weird some of them is default some of them is null